From fa097a7d7ff6417add8c1129832c49c6a2ff8bc0 Mon Sep 17 00:00:00 2001
From: Daniel Micay <daniel.micay@grapheneos.org>
Date: Fri, 2 Feb 2024 18:54:40 -0500
Subject: [PATCH] run full explicit GC in SystemUI and system_server after

Ported to AXP.OS lineage-22.2 (rhodep fork) from GrapheneOS b53b0d286cec.

Change-Id: I7017d3f0d5703a69be9bc0b99a1322eb69863704
---
 .../com/android/systemui/keyguard/KeyguardViewMediator.java | 3 +++
 .../server/policy/keyguard/KeyguardStateMonitor.java        | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
index b791c8f45..b0c82063a 100644
--- a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
+++ b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
@@ -3117,6 +3117,9 @@ public class KeyguardViewMediator implements CoreStartable, Dumpable,
             userActivity();
             mUpdateMonitor.setKeyguardGoingAway(false);
             mKeyguardViewControllerLazy.get().setKeyguardGoingAwayState(false);
+            System.gc();
+            System.runFinalization();
+            System.gc();
             mShowKeyguardWakeLock.release();
         }
         mKeyguardDisplayManager.show();
diff --git a/services/core/java/com/android/server/policy/keyguard/KeyguardStateMonitor.java b/services/core/java/com/android/server/policy/keyguard/KeyguardStateMonitor.java
index c0aa8aeff..6e086c6c2 100644
--- a/services/core/java/com/android/server/policy/keyguard/KeyguardStateMonitor.java
+++ b/services/core/java/com/android/server/policy/keyguard/KeyguardStateMonitor.java
@@ -89,6 +89,12 @@ public class KeyguardStateMonitor extends IKeyguardStateCallback.Stub {
         mIsShowing = showing;
 
         mCallback.onShowingChanged();
+
+        if (showing) {
+            System.gc();
+            System.runFinalization();
+            System.gc();
+        }
     }
 
     @Override // Binder interface
-- 
2.43.0

